x-pack/legacy/plugins/infra/public/components/waffle/node_context_menu.tsx

```diff
@@ -43,83 +48,116 @@ export const NodeContextMenu = ({
 }: Props) => {
   const uiCapabilities = useKibana().services.application?.capabilities;
   const inventoryModel = findInventoryModel(nodeType);
+  const nodeDetailFrom = currentTime - inventoryModel.metrics.defaultTimeRangeInSeconds * 1000;
   // Due to the changing nature of the fields between APM and this UI,
   // We need to have some exceptions until 7.0 & ECS is finalized. Reference
   // #26620 for the details for these fields.
   // TODO: This is tech debt, remove it after 7.0 & ECS migration.
   const apmField = nodeType === 'host' ? 'host.hostname' : inventoryModel.fields.id;
 
-  const nodeLogsMenuItem = {
-    name: i18n.translate('xpack.infra.nodeContextMenu.viewLogsName', {
-      defaultMessage: 'View logs',
+  const showDetail = inventoryModel.crosslinkSupport.details;
+  const showLogsLink =
+    inventoryModel.crosslinkSupport.logs && node.id && uiCapabilities?.logs?.show;
+  const showAPMTraceLink =
+    inventoryModel.crosslinkSupport.apm && uiCapabilities?.apm && uiCapabilities?.apm.show;
+  const showUptimeLink =
+    inventoryModel.crosslinkSupport.uptime && (['pod', 'container'].includes(nodeType) || node.ip);
+
+  const nodeLogsMenuItem: EuiListGroupItemProps = {
```

Comment:
Would it make sense to re-export the `EuiListGroupItemProps` from `./action_menu` as `SectionLinkProps` to avoid breaking the abstraction in case we want to modify the props list in the future?